heareval/predictions/task_predictions.py

```diff
@@ -904,6 +981,125 @@ def hparams_to_json(hparams):
+                "train+valid": train_folds + [valid_fold],
+                "test": [test_fold],
+            }
+        )
+        assert not set(train_folds).intersection(
+            {test_fold, valid_fold}
+        ), "Train folds are not distinct from the dev and the test folds"
+
+    return all_data_splits
+
+
+def aggregate_test_results(results: Dict[str, Dict[str, float]]) -> Dict[str, float]:
+    """
+    Aggregates test results over folds by mean and standard deviation
+    """
+    results_df = pd.DataFrame.from_dict(results, orient="index")
+    aggregate_results = {}
+    for score in results_df:
+        aggregate_results[f"{score}_mean"] = results_df[score].mean()
+        aggregate_results[f"{score}_std"] = results_df[score].std()
+
+    return aggregate_results
+
+
+def get_splits_from_metadata(metadata: Dict) -> List[Dict[str, List[str]]]:
+    """
+    Extracts the splits for training from the task metadata. If there are folds
+    present then this creates a set of k splits for each fold.
+    """
+    data_splits: List[Dict[str, List[str]]]
+    if metadata["split_mode"] == "trainvaltest":
+        # There are train/validation/test splits predefined. These are the only splits
+        # that will be considered during training and testing.
+        data_splits = [
+            {
+                "train": ["train"],
+                "valid": ["valid"],
+                "train+valid": ["train" + "valid"],
+                "test": ["test"],
+            }
+        ]
+    elif metadata["split_mode"] in ["new_split_kfold", "presplit_kfold"]:
+        folds = metadata["splits"]
+        # Folds should be a list of strings
+        assert all(isinstance(x, str) for x in folds)
+        # If we are using k-fold cross-validation then get a list of the
+        # splits to test over. This expects that k data folds were generated
+        # during pre-processing and the names of each of these folds is listed
+        # in the metadata["folds"] variable.
+        data_splits = data_splits_from_folds(folds)
+    else:
+        raise AssertionError(
+            f"Unknown split_mode: {metadata['split_mode']} in task metadata."
+        )
+
+    return data_splits
+
+
+def sort_grid_points(grid_point_results: List[GridPointResult]) -> None:
+    """
+    Sort grid point results in place, so that the first result
+    is the best.
+    """
+    # TODO: Assert all score modes are the same?
+    mode = grid_point_results[0].score_mode
+    # Pick the model with the best validation score
+    grid_point_results.sort(key=lambda g: -g.validation_score)
+    if mode == "max":
+        pass
+    elif mode == "min":
+        grid_point_results.reverse()
+    else:
+        raise ValueError(f"mode = {mode}")
+
+
+def print_scores(grid_point_results: List[GridPointResult], embedding_path: Path):
+    sort_grid_points(grid_point_results)
+    for g in grid_point_results:
+        print(
+            json.dumps(
```

Comment:
Why not make a GridPointResult.str or GridPointResult.json() method, to simplify?